frontend/src/app/subject-comments.service.ts

```diff
@@ -0,0 +1,176 @@
+                    return null;
+                }
+            })
+            .catch(e => {
+                console.log(`Remove thumb up from comment (id:${comment}) request failed. Caught exception: ${e}`);
+                return null;
+            });
+    }
+
+    remThumbDown(university: number, subject: number, comment: number): Promise<SubjectComment>{
+        return this.http
+            .get(`http://localhost:8080/api/universities/university/${university}/subjects/subject/${subject}/comments/comment/${comment}/remThumbDown`)
+            .toPromise()
+            .then(response => {
+                if(response.status == 200){
+                    return response.json() as SubjectComment;
+                } else {
+                    console.log(`Unable to remove thumb down from comment (id:${comment}) for subject id: ${subject}`);
+                    return null;
+                }
+            })
+            .catch(e => {
+                console.log(`Remove thumb down from comment (id:${comment}) request failed. Caught exception: ${e}`);
+                return null;
+            });
+    }
+
+    remThumbs(university: number, subject: number, comment: number): Promise<SubjectComment>{
+        return this.http
+            .get(`http://localhost:8080/api/universities/university/${university}/subjects/subject/${subject}/comments/comment/${comment}/remThumbs`)
+            .toPromise()
+            .then(response => {
+                if(response.status == 200){
+                    return response.json() as SubjectComment;
+                } else {
+                    console.log(`Unable to remove all thumbs from comment (id:${comment}) for subject id: ${subject}`);
+                    return null;
+                }
+            })
+            .catch(e => {
+                console.log(`Remove all thumbs from comment (id:${comment}) request failed. Caught exception: ${e}`);
+                return null;
+            });
+    }
+
+    flag(university: number, subject: number, comment: number): Promise<SubjectComment>{
+        return this.http
+            .get(`http://localhost:8080/api/universities/university/${university}/subjects/subject/${subject}/comments/comment/${comment}/flag`)
+            .toPromise()
+            .then(response => {
+                if(response.status == 200){
+                    return response.json() as SubjectComment;
+                } else {
+                    console.log(`Unable to flag comment (id:${comment}) for subject id: ${subject}`);
+                    return null;
+                }
+            })
+            .catch(e => {
+                console.log(`Flag comment (id:${comment}) request failed. Caught exception: ${e}`);
+                return null;
+            });
+    }
+
+    unFlag(university: number, subject: number, comment: number): Promise<SubjectComment>{
```

Comment:
This looks like it will work, and you don't need to change it. But as a tip, programmers are lazy. You've got code duplicated here, you should try and generalise it. So instead of having all this boilerplate around every request inside your methods, you could make a method that is just basically something like:
```
getSubject(url: string, errorMessage: string) {
  http.get(url)
  .toPromise()
  .then(response => {
    if(response.status == 200){
      return response.json() as SubjectComment;
    } else {
      console.log(errorMessage);
      return null;
    }
  })
  .catch(e => {
    console.log(errorMessage);
    return null;
})
```

Reply:
added :D